addons/point_of_sale/static/src/js/models.js

```diff
@@ -221,7 +221,13 @@ exports.PosModel = Backbone.Model.extend({
         fields: ['id', 'journal_ids','name','user_id','config_id','start_at','stop_at','sequence_number','login_number'],
         domain: function(self){ return [['state','=','opened'],['user_id','=',session.uid]]; },
         loaded: function(self,pos_sessions){
-            self.pos_session = pos_sessions[0]; 
+            self.pos_session = pos_sessions[0];
+
+            // reset the cashier to the current user if session is new
+            if (self.db.load('pos_session_id') !== self.pos_session.id) {
+                self.set_cashier(self.user);
+            }
+            self.db.save('pos_session_id', self.pos_session.id);
```

Comment:
Theoretically a dict `session_id -> cashier_user_id` is more "correct" because in the db there can be data (mainly [unpaid orders](https://github.com/odoo/odoo/blob/896e3f4c1eace0c1ad3e3b472fa373e62d7cddb5/addons/point_of_sale/static/src/js/models.js#L652)) for multiple sessions. However, since it's a rare case I think that the added complexity of managing this is not worth the effort (you'd have to somehow clear entries from this dict so that it doesn't indefinitely grow). When this rare case occurs with your solution all that will happen is the user of the session will be reset to the default for that session, not a big issue.